src/renderers/WebGLRenderer.js

```diff
@@ -1305,7 +1307,8 @@ class WebGLRenderer {
 
 						if ( sortObjects ) {
 
-							_vector3.setFromMatrixPosition( object.matrixWorld )
+							_vector3.setFromMatrixPosition( object.matrixWorld );
+							_vector4.copy( _vector3 )
```

Comment:
I'm wondering if we should add `Vector4.setMatrixFromPosition` or `Vector4.setFromMatrixColumn` so we can facilitate this kind of use more clenaly.

cc @mrdoob

Reply:
Let's focus on getting this PR correct, first. If we sort in camera space, this is a non-issue.

Reply:
Can you explain how the previous commits were _not_ correct? It was sorting in the camera clip space before (w divide) which is arguably more correct since this is where the triangle projection and rasterization is actually happening. The fundamental issue is that dividing by w can flip the sign of the object origin.

This is what was done before the commits were overwritten:

> ```js
> _vec3.setFromMatrixPosition( object.matrixWorld );
> _vec4.copy( _vec3 ).applyMatrix4( _projScreenMatrix );
> // sort on _vec4.z ...
> ```

Reply:
One comment on the code here is that if we don't want to add a new function for vector4 to get the translation column is that this can still be written in one line like so:

```js
_vec4.set( 0, 0, 0, 1 )
  .applyMatrix4( object.matrixWorld )
  .applyMatrix4( _projScreenMatrix ); 
```

Reply:
Having a `Vector4.setFromMatrixPosition` makes sense in my mind, at least. If we don't want to add that, I'd say we should avoid the extra vector-matrix multiplication (which due to formatting still adds a line..), and just use the copy function.

Reply:
Added `Vector4.setMatrixFromPosition` sounds good to me 👍